### PR TITLE
[irods/irods6369] update iscan help text for clarity (main)

### PR DESCRIPTION
**Pull request by trel**

### CODE REVIEW

**Review by korydraughn — COMMENTED** [review]
The current changes are fine. I think an example or two would be good here.

For instance, it's not clear to me **what** is being checked. Is `iscan` verifying there's a row in the catalog that matches the absolute path for the local file or is it verifying only the final path element exists somewhere in the catalog?

What output should the user expect?

**Review by alanking — COMMENTED** [review]
OP in irods/irods#6369 requested a usage example. Y'know, if you're into it.

**Comment by trel** [comment]
added some examples.

File: src/iscan.cpp [hunk 1]
```diff
@@ -100,23 +100,40 @@ usage() {
...
         "Options are:",
-        " -r  recursive - scan local subdirectories or subcollections",
+        " -r  recursive - scan local subdirectories or logical subcollections",
         " -h  this help",
-        " -d  scan data objects in iRODS (default is scan local objects)",
+        " -d  scan data objects in iRODS (default is scan local files/directories)",
+        " ",
+        "Examples:",
+        "$ iscan /var/lib/irods/notincatalog",
+        "/var/lib/irods/notincatalog is not registered in iRODS",
```
Comment (korydraughn) [inline src/iscan.cpp:122]: Inconsistent naming between `notincatalog` and `not_a_dataobject` (on L131).

**Review by korydraughn — COMMENTED** [review]
Examples are good, but I think they need to include words explaining what they are doing. It wasn't obvious what they were attempting to convey or whether they were building on each other.

File: src/iscan.cpp (hunk 1 above)
Comment (trel) [reply src/iscan.cpp:122]: ah, updated.

File: src/iscan.cpp [hunk 2]
```diff
@@ -100,23 +100,49 @@ usage() {
...
+        "Could not find the requested data object or collection in iRODS.",
+        " ",
+        "# recursively scan a relative logical path for missing physical paths",
+        "$ iscan -rd scanme",
+        "Physical file /var/lib/irods/Vault/home/rods/scanme/does_not_exist on server irods.example.org is missing, corresponding to iRODS object /tempZone/home/rods/scanme/does_not_exist",
+        "Physical file /var/lib/irods/Vault/home/rods/scanme/missing on server irods.example.org is missing, corresponding to iRODS object /tempZone/home/rods/scanme/missing",
+        " ",
+        "# rodsadmin required to scan logical paths",
+        "$ iscan -d test.txt",
+        "User must be a rodsadmin to scan iRODS data objects.",
```
Comment (korydraughn) [inline src/iscan.cpp:145]: Does this mean `iscan` cannot be used by rodsusers/groupadmins?

Comment (trel) [reply src/iscan.cpp:145]: nonadmins can scan physical paths, but not logical paths

Comment (korydraughn) [reply src/iscan.cpp:145]: Got it.

**Review by korydraughn — APPROVED** [review]
I think this is in a good place.

Feel free to squash/pound.

**Comment by trel** [comment]
squashed

**Review by alanking — APPROVED** [review]

**Comment by korydraughn** [comment]
Did you ever see a successful build of this?

**Comment by trel** [comment]
not yet - and need my main machine available for other things.   happy to receive confirmation if you want to compile.

**Comment by korydraughn** [comment]
Ok. Will do.

**Comment by alanking** [comment]
Built for me, fwiw

**Comment by korydraughn** [comment]
And the help text for iscan looks good?

**Comment by alanking** [comment]
Looks great:
```bash
$ iscan -h
Usage: iscan [-rhd] srcPhysicalFile|srcPhysicalDirectory|srcDataObj|srcCollection
 
If the input is a local physical data file or a local directory,
iscan checks if the named content is registered in iRODS.
A full physical path must be used for local files and directories.
 
If the input is a logical path representing an iRODS data object
or an iRODS collection, iscan checks if the associated registered
physical file entries exist on the data servers.
Scanning data objects and collections may only be performed by a rodsadmin.
 
If the operation is successful, nothing will be output
and 0 will be returned.
 
Options are:
 -r  recursive - scan local subdirectories or logical subcollections
 -h  this help
 -d  scan data objects in iRODS (default is scan local files/directories)
 
Examples:
# scan a single physical path to check its presence in the catalog
$ iscan /var/lib/irods/not_in_catalog
/var/lib/irods/not_in_catalog is not registered in iRODS
 
# recursively scan a physical path to check for unregistered local files
$ iscan -r /var/lib/irods/Vault/home/rods/scanned
/var/lib/irods/Vault/home/rods/scanned/does_not_exist2 is not registered in iRODS
/var/lib/irods/Vault/home/rods/scanned/does_not_exist1 is not registered in iRODS
 
# recursively scan a physical path to check for unregistered local files
$ iscan -r /data/projects/myproject
remote addresses: 127.0.0.1 ERROR: scanObj: /data/projects/myproject does not exist
 
# check that a single relative logical path exists in the catalog
$ iscan -d not_a_dataobject
Could not find the requested data object or collection in iRODS.
 
# recursively scan a relative logical path for missing physical paths
$ iscan -rd scanme
Physical file /var/lib/irods/Vault/home/rods/scanme/does_not_exist on server irods.example.org is missing, corresponding to iRODS object /tempZone/home/rods/scanme/does_not_exist
Physical file /var/lib/irods/Vault/home/rods/scanme/missing on server irods.example.org is missing, corresponding to iRODS object /tempZone/home/rods/scanme/missing
 
# rodsadmin required to scan logical paths
$ iscan -d test.txt
User must be a rodsadmin to scan iRODS data objects.

iRODS Version 4.90.0                iscan
```

**Review by korydraughn — APPROVED** [review]
Let's pound it!

**Comment by trel** [comment]
#'d

**Pull request merged**